builtin/providers/postgresql/resource_postgresql_database_funcs.go

```diff
@@ -0,0 +1,82 @@
+package postgresql
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/lib/pq"
+)
+
+func resourcePostgresqlDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*sql.DB)
+	dbName := d.Get("name").(string)
+	dbOwner := d.Get("owner").(string)
+	var dbOwnerCfg string
+	if dbOwner != "" {
+		dbOwnerCfg = fmt.Sprintf("WITH OWNER=%s", pq.QuoteIdentifier(dbOwner))
+	} else {
+		dbOwnerCfg = ""
+	}
+
+	query := fmt.Sprintf("CREATE DATABASE %s %s", pq.QuoteIdentifier(dbName), dbOwnerCfg)
+	_, err := conn.Query(query)
+	if err != nil {
+		return fmt.Errorf("Error creating postgresql database: %s", err)
+	}
+
+	d.SetId(dbName)
+
+	return nil
+}
+
+func resourcePostgresqlDatabaseDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*sql.DB)
+	dbName := d.Get("name").(string)
+
+	query := fmt.Sprintf("DROP DATABASE %s", pq.QuoteIdentifier(dbName))
+	_, err := conn.Query(query)
+	if err != nil {
+		return err
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourcePostgresqlDatabaseRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*sql.DB)
+	dbName := d.Get("name").(string)
+
+	var owner string
+	err := conn.QueryRow("SELECT pg_catalog.pg_get_userbyid(d.datdba) from pg_database d WHERE datname=$1", dbName).Scan(&owner)
+	switch {
+	case err == sql.ErrNoRows:
+		d.SetId("")
+		return nil
+	case err != nil:
+		return fmt.Errorf("Error reading info about database: %s", err)
+	default:
+		d.Set("owner", owner)
+		return nil
+	}
+}
+
+func resourcePostgresqlDatabaseUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*sql.DB)
+	dbName := d.Get("name").(string)
+
+	if d.HasChange("owner") {
+		owner := d.Get("owner").(string)
+		if owner != "" {
```

Comment:
If a database is initially created with an explicit `owner`, and then that argument is removed from the configuration, I think this will cause a problem because Terraform's planning/diffing code will think it needs to set the owner to the empty string but `terraform apply` (which will run this code) won't actually make that change.

I think making the field `Computed` is the right answer for this, too. That will mean that when the `owner` argument is removed from the config Terraform should show the diff as being something like `"foo" -> (computed)", and then it will populate the state with whatever existing owner is obtained by the call to `Read` at the end of this function.